cmd/clusterctl/pkg/client/cluster/inventory.go

```diff
@@ -53,7 +53,7 @@ type InventoryClient interface {
 	Create(clusterctlv1.Provider) error
 
 	// List returns the inventory items for all the provider instances installed in the cluster.
-	List() ([]clusterctlv1.Provider, error)
+	List() (*clusterctlv1.ProviderList, error)
```

Comment:
Switching from `[]clusterctlv1.Provider` to `clusterctlv1.ProviderList` so I can drop the double List / list function